### ISSUE TITLE
Memory Leak when processing a cgImage

### ISSUE THREAD

**Issue by Mattijah** [SwiftyTesseract/SwiftyTesseract#50]
Memory does not seem to be released when processing `UIImage` created with a `CGImage`, **which was created off of a** `CIImage`.

**Example**:
```swift
public func captureOutput(_ output: AVCaptureOutput, didOutput sampleBuffer: CMSampleBuffer, from connection: AVCaptureConnection) {
    let context = CIContext.init(options: nil)
    let imageBuffer = CMSampleBufferGetImageBuffer(sampleBuffer)!
    let ciImage = CIImage(cvPixelBuffer: imageBuffer)
    let cgImage = context.createCGImage(ciImage, from: ciImage.extent)!
    let image = UIImage(cgImage: cgImage)

    tesseract.performOCR(on: image) { _ in } // Called repeatedly
}
```

**Instruments**:
![mem](https://user-images.githubusercontent.com/3887042/74597232-fc198b80-505b-11ea-9050-2948a6661582.png)

*I know I could also create an image with a `CIImage` (`UIImage(ciImage: ciImage)`) and this would work well, but even something like above should.

**Comment by Steven0351** [comment]
I'm looking into this, but as of yet I'm not able to reproduce your results. I've run both Leaks and Activity Monitor from Instruments. The max memory being used I've witnessed via Activity Monitor was ~45 MB and there were no leaks according to the leaks tool. I've also not seen anything of concern in the Time Profiler. Could you provide a minimal full example (that I can run on my device) that you are seeing these results?

**Comment by Mattijah** [comment]
[Here](https://drive.google.com/file/d/1B7YikmT4eKMhfPCon5FSl_U_cv4HuWrs/view?usp=sharing) is a demo app. All of the logic is located in the `AppDelegate` (please don't pay much attention to that it's just quickly put together). I have also added a comment in there. Furthermore, I have only used **Allocations** to check this leak, but you should be able to see the memory increase even in Xcode itself.

![mem_increase](https://user-images.githubusercontent.com/3887042/74613470-f8930c80-510e-11ea-80e0-4b2c331b8427.png)

**Comment by Steven0351** [comment]
So I definitely am able to reproduce this now thanks to the project. Also, huge thanks for pointing out the issue seeming to be only with that specific filter. Based on some preliminary digging around it seems to be an issue with Tesseract 4.0.0 and appears to not be an issue with Tesseract 4.1.0. I'm going to do some more validating to make sure, but this may be as simple as shipping an updated version of the bundled tesseract libs

**Comment by Steven0351** [comment]
I have found no reason to suspect that this isn't an issue with the version of Tesseract and upgrading to 4.1.x will be the fix for this. One little setback however, is that the current way these are built is on my local and committed to the repo because of how long the build takes for all the various architectures. I'm on Catalina and the dropping of 32-bit support seems to also extend to not even being able to compile 32-bit code at all. I've been looking into trying to get a Mojave VM running to get around that, but so far the Mac App Store does not seem to be too keen on allowing that.

**Comment by Mattijah** [comment]
Thank you for looking into it!

> the issue seeming to be only with that specific filter

I'm sure I have also come across this issue in `captureOutput` as mentioned in my initial comment (without using any filters). The underlying problem may however be the same anyway.

> 32 bit

32 bit is only supported on iOS 9 / iOS 10 I believe. These version have less than 1.5% market share, so dropping support should not be a major problem in my opinion.

**Comment by Steven0351** [comment]
This has been addressed with version 3.0.0. The previous `performOCR(on:completionHandler:)` method has been deprecated and will be removed in a future release, however it is still available for the time being.
